updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,220 @@
+During an OpenShift update all pools are updated concurrently.
+
+Nodes within a MCP are updated by cordoning and draining up to the specified `maxUnavailable` number of nodes (if required).
+Once drained the Machine Config Daemon applies MachineConfig changes which may include updating the OS and the host is rebooted.
+The MCP gives us the ability to pause a pool and it will make sure that the OS is not updated/rebooted during the update process.
+We can create more than one customized machine config pool out of the worker pool, which will give us a more control over
+the sequence in which the nodes are updated.
+
+[NOTE]
+----
+Creating custom MCP using master nodes is not supported.
+Machine config operator (responsible for updating the nodes) will ignore the custom MCP created out of the master nodes.
+This is required to make sure control plane nodes remain stable.
+----
+
+An OpenShift or Kubernetes cluster is highly available by design and there are many Kubernetes features
+(e.g. pod disruption budget, pod affinity, health checks, replicas) which can make the application highly available in case of node failures.
+
+However there might be some secenarios where a controlled rollout of upgrade to the worker nodes is desired.
+This helps to make sure the mission critical application stays available during the whole update process even if the upgrade process causes a failure.
+It is also useful in scenarios where we can not have a single maintenance window to complete the whole update process.
+
+The slow rollout of a new OpenShift version to worker nodes can be characterized as a canary release i.e. we can
+control the rollout of worker nodes by controlling the machine config pools. After the update of the first MCP, the
+application compatibility can be verfied and then the whole fleet can be updated gradually to the new version.
+
+
+== Workflow for updating worker nodes with canary rollout
+
+. Create MCPs out of the worker pool. The number of nodes in each MCP depends on a few factors like maintaince window duration for each MCP and the amount of reserve capacity (extra worker nodes) present.
++
+[NOTE]
+====
+In case of a failure i.e. if the MCP with the new version does not work as expected with the applications, the nodes in the pool can be cordoned and drained. So that the application will not run on these nodes and the extra capacity will help to maintain the quality of service of the applications.
+====
++
+. Pause the MCPs created in the previous step.
++
+[NOTE]
+====
+Pausing the MCP will also pause the kube-apiserver-to-kubelet-signer automatic CA certificates rotation.
+You need to unpause these MCPs before the next automatic CA certificate renewal is scheduled.
+See the link:https://access.redhat.com/articles/5651701[article] to findout how much time you have before the next automatic CA certificate renewal happens.
```

Comment:
Linked KCS article doesn't clearly specify that you have to apply cert at 292 days on node. It just says, cert is renewed at 292 days and expiry is 365 days. To avoid confusion, I think we should clearly specify in this doc that new cert should be allowed to be applied on node when renewal occurs i.e. 292 days and then post 365 days, it can be unpaused again to remove old certs. Otherwise, cluster will hit the issue, see https://bugzilla.redhat.com/show_bug.cgi?id=1970329#c25

Reply:
With the latest update I have changed the language to include what you suggested. PATL